File: car-hunter-hamburg/backend/app/schemas/car.py
"""
Schémas Pydantic : forme des données échangées par l'API.
Séparés des modèles SQLAlchemy pour ne jamais exposer la structure
de la base de données telle quelle (bonne pratique de sécurité/évolutivité).
"""
from datetime import datetime
from typing import Optional
from pydantic import BaseModel, ConfigDict


class CarBase(BaseModel):
    brand: str
    model: str
    year: Optional[int] = None
    price: int
    mileage_km: int
    fuel_type: Optional[str] = None
    transmission: Optional[str] = None
    location: Optional[str] = None
    distance_from_hamburg_km: Optional[float] = None
    url: str


class CarOut(CarBase):
    model_config = ConfigDict(from_attributes=True)

    id: int
    source: str
    power_kw: Optional[int] = None
    owners_count: Optional[int] = None
    consumption_l_100km: Optional[float] = None
    image_url: Optional[str] = None
    score: Optional[int] = None
    ai_analysis: Optional[str] = None
    is_active: bool
    date_added: datetime
    date_last_seen: datetime


class CarListResponse(BaseModel):
    total: int
    items: list[CarOut]

File: car-hunter-hamburg/frontend/src/types/car.ts
export interface Car {
  id: number;
  source: string;
  brand: string;
  model: string;
  year: number | null;
  price: number;
  mileage_km: number;
  fuel_type: string | null;
  transmission: string | null;
  location: string | null;
  distance_from_hamburg_km: number | null;
  url: string;
  power_kw: number | null;
  owners_count: number | null;
  consumption_l_100km: number | null;
  image_url: string | null;
  score: number | null;
  ai_analysis: string | null; // JSON stringifié — voir parseAiAnalysis()
  is_active: boolean;
  date_added: string;
  date_last_seen: string;
}

export interface AiAnalysis {
  label: string;
  recommendation: string;
  strengths: string[];
  watch: string[];
}

export function parseAiAnalysis(car: Car): AiAnalysis | null {
  if (!car.ai_analysis) return null;
  try {
    return JSON.parse(car.ai_analysis) as AiAnalysis;
  } catch {
    return null;
  }
}

export interface CarListResponse {
  total: number;
  items: Car[];
}
